src/main/resources/native/android/c/launcher.c

```diff
@@ -60,6 +80,16 @@ JNIEXPORT void JNICALL Java_com_gluonhq_helloandroid_MainActivity_nativeSetSurfa
     LOGE(stderr, "native setSurface Ready, native window at %p\n", window);
 }
 
+JNIEXPORT void JNICALL Java_com_gluonhq_helloandroid_MainActivity_nativeSetDataDir
+(JNIEnv *env, jobject that, jstring jdir) {
+    const char *cdir = (*env)->GetStringUTFChars(env, jdir, 0);
+    int len = strlen(cdir);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4782774)